dynamic_rest/content_type_field.py

```diff
@@ -0,0 +1,73 @@
+import traceback
+
+from dynamic_rest.fields import DynamicField
+from dynamic_rest.routers import DynamicRouter
+
+
+class DynamicContentTypeField(DynamicField):
+
+    def __init__(
+        self,
+        object_field=None,
+        type_field=None,
+        id_field=None,
+        *args, **kwargs
+    ):
+        # NOTE: object_field should be same as `source`, except `source`
+        #       isn't reliably available until this field is bound().
+        if not object_field:
+            raise Exception(
+                'DynamicContentTypeField requires `object_field` kwarg'
+            )
+
+        # Infer type field and id field from object_field if not explicitly set
```

Comment:
Yeah, this is mostly a side-effect of trying to do this in `__init__` where `parent` isn't available yet, so the model field isn't accessible. I think doing all this in `bind()` is pretty safe, since fields should be bound before any of the other methods in this class get called. I'll try moving it there and see how that looks.